pymc/distributions/multivariate.py

```diff
@@ -2153,3 +2160,120 @@ def logp(value, mu, W, alpha, tau):
+
+    def __call__(self, alpha=1.0, K=10, size=None, **kwargs):
+        return super().__call__(alpha, K, size=size, **kwargs)
+
+    def _infer_shape(self, size, dist_params, param_shapes=None):
+        alpha, K = dist_params
+
+        size = tuple(size)
+
+        return size + (K + 1,)
+
+    @classmethod
+    def rng_fn(cls, rng, alpha, K, size):
+        if K < 0:
+            raise ValueError(f"K needs to be a positive integer, but K = {K}.")
+
+        if np.ndim(alpha) > 0:
+            raise ValueError("The concentration parameter needs to be a scalar.")
+
+        if size is None:
+            size = (K,)
+        elif isinstance(size, int):
+            size = (size,) + (K,)
+        else:
+            size = tuple(size) + (K,)
+
+        betas = rng.beta(1, alpha, size=size)  # adapt this to vector alpha
+
+        sticks = np.concatenate(
+            (
+                np.ones(shape=(size[:-1] + (1,))),
+                np.cumprod(1 - betas[..., :-1], axis=-1),
+            ),
+            axis=-1,
+        )
+
+        weights = sticks * betas
+        weights = np.concatenate(
+            (weights, 1 - weights.sum(axis=-1)[..., np.newaxis]),
+            axis=-1,
+        )
+
+        return weights
+
+
+stickbreakingweights = StickBreakingWeightsRV()
+
+
+class StickBreakingWeights(Continuous):
+    rv_op = stickbreakingweights
+
+    def __new__(cls, name, *args, **kwargs):
+        kwargs.setdefault("transform", transforms.simplex)
+        return super().__new__(cls, name, *args, **kwargs)
+
+    @classmethod
+    def dist(cls, alpha, K, *args, **kwargs):
+        alpha = at.as_tensor_variable(floatX(alpha))
+        K = at.as_tensor_variable(intX(K))
+
+        if alpha.ndim > 0:
+            raise ValueError("alpha must be a scalar.")
+
+        if K.ndim > 0:
+            raise ValueError("K must be a scalar.")
+
+        assert_negative_support(alpha, "alpha", "StickBreakingWeights")
+        assert_negative_support(K, "K", "StickBreakingWeights")
+
+        return super().dist([alpha, K], **kwargs)
+
+    def get_moment(rv, size, alpha, K):
+        moment = (alpha / (1 + alpha)) ** at.arange(K)
+        moment *= 1 / (1 + alpha)
+        moment = at.concatenate([moment, [(alpha / (1 + alpha)) ** K]], axis=-1)
+        if not rv_size_is_none(size):
+            moment_size = at.concatenate(
+                [
+                    size,
+                    [
+                        K + 1,
+                    ],
+                ]
+            )
+            moment = at.full(moment_size, moment)
+
+        return moment
+
+    def logp(value, alpha, K):
+        logp = -at.sum(
+            at.log(
+                at.cumsum(
+                    value[..., ::-1],
+                    axis=-1,
+                )
+            ),
+            axis=-1,
+        )
+        logp += -(K - 1) * betaln(1, alpha)
+        logp += alpha * at.log(value[..., -1])  # check this
+
+        return check_parameters(
```

Comment:
We need tests for the logp parameter checks and multivariate logp `value`s.

You can check `test_dirichlet_multinomial_vectorized` and `test_dirichlet_multinomial_invalid` for examples of both.

If you add a test that uses `check_logp` it will test for invalid scalar parameters `alpha` and `k` automatically. You would only need manual tests for the `value`